solr/solrj/src/java/org/apache/solr/common/cloud/DocCollection.java

```diff
@@ -432,6 +484,34 @@ public String getPolicyName() {
     return policy;
   }
 
+  public boolean isPerReplicaState() {
+    return Boolean.TRUE.equals(perReplicaState);
+  }
+
+  public PerReplicaStates getPerReplicaStates() {
+    return perReplicaStates;
+  }
+
+  /**
+   * for internal use only
+   */
+  public PerReplicaStates getPerReplicaStates(SolrZkClient zkClient)  {
```

Comment:
Does this method need to be synchronized?
This is not a simple accessor. It mutates local state on read. That seems dangerous?

Reply:
It's documented as `for internal use only`. But this is a `public` method. Who are the users and when should this be used?

Reply:
This method must go. It doesn't belong here